### PR TITLE
Add roadmap doc & diagram

### PR DESCRIPTION
**Pull request by karlfloersch**
**Description**
Adds a high level roadmap diagram & document which is intended to communicate:

1. The different streams of work / research.
2. The different stages of iteration that are possible as we complete each component as well as what is blocking.
3. The relationship between the near term work that we are doing to make sure Geth is as minimal as possible and the future dream of a natively integrated EVM EE.

If successful this document should help tell the story of: 1) minimize diff, 2) integrate merge api, 3) integrate a 1:1 geth fraud proof, 4) eth2 phase 1 massive scale, 5) full Ethereum L2 capabilities 

Note: this also moves the assets into an `assets` directory.

### CODE REVIEW

**Comment by protolambda** [comment]
Some suggested changes in naming:
- "eth2 phase 1" -> sharding
- "EE" -> Execution Engine, what is currently an eth1 node. (~2 years ago we used "EE" for "Execution Environment", an ewasm or plugin kind of thing, that changed a lot)

**Comment by protolambda** [comment]
"Opcode single step executor" -> "single step verifier"

**Comment by karlfloersch** [comment]
Perfect updated the naming!

File: roadmap.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+Each component will produce incremental and independent releases, each driving closer to unification and Optimistic Ethereum nirvana.
+
+## Summary of each component
+
+### Optimism Mainnet
+
+Optimism mainnet is the first Optimistic Ethereum network. It serves as:
+
+1. A testbed for protocol changes; and
```
Comment (lightclient) [inline roadmap.md:21]: I think selling L2s as a testbed for protocol changes is slippery slope. As L2s accrue value, I expect users will become more and more cautious of changes.

File: roadmap.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+### Fraud Proof
+
+The fraud proof contracts and infrastructure are a pluggable set of EVM smart contracts, off-chain witness generators, challenge agents, and more. The near term fraud proof design is a 1:1 implementation of the Ethereum protocol written in the EVM. In the medium to long term, after statelessness is introduced to Ethereum (see the next project), the fraud proof should become entirely native so that Ethereum clients remain fully compatible with layer 2.
+
+### eth1.x stateless clients
+
+State expiry and stateless clients are key to solving the Ethereum state bloat problem **and** for native EVM fraud proofs. Once stateless clients are ready we can transition the Optimistic Ethereum fraud proof to be native to Geth
+
+Note: Although critical to the OE roadmap, this project is tracked [external](https://github.com/ethereum) to this repository.
```
Comment (lightclient) [inline roadmap.md:34]:
```suggestion
Note: Although critical to the OE roadmap, this project is tracked [external](https://github.com/ethereum/stateless-ethereum-specs/) to this repository.
```

I think this is stateless repo?

File: roadmap.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+### eth2
+
+The eth2 merge API and sharding will allow for native integration into execution engines (eg. Geth) as well as massive scalability ([~100k TPS](https://vitalik.ca/general/2021/01/05/rollup.html)) by greatly increasing Ethereum's data availability bandwidth.
+
+Note: Although critical to the OE roadmap, this project is tracked [external](https://github.com/ethereum) to this repository.
```
Comment (lightclient) [inline roadmap.md:40]:
```suggestion
Note: Although critical to the OE roadmap, this project is tracked [external](https://github.com/ethereum/consensus-specs) to this repository.
```

File: roadmap.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+This network will start out as being a rollup using eth1 as a data availability engine, and will migrate once sharding is ready.
+
+### Fraud Proof
+
+The fraud proof contracts and infrastructure are a pluggable set of EVM smart contracts, off-chain witness generators, challenge agents, and more. The near term fraud proof design is a 1:1 implementation of the Ethereum protocol written in the EVM. In the medium to long term, after statelessness is introduced to Ethereum (see the next project), the fraud proof should become entirely native so that Ethereum clients remain fully compatible with layer 2.
+
+### eth1.x stateless clients
+
+State expiry and stateless clients are key to solving the Ethereum state bloat problem **and** for native EVM fraud proofs. Once stateless clients are ready we can transition the Optimistic Ethereum fraud proof to be native to Geth
```
Comment (lightclient) [inline roadmap.md:32]:
```suggestion
State expiry and stateless clients are key to solving the Ethereum state bloat problem **and** for native EVM fraud proofs. Once stateless clients are ready we can transition the Optimistic Ethereum fraud proof to a native component of the EVM.
```

Comment (lightclient) [reply roadmap.md:32]: IMO, if OE *is* Ethereum we should be inclusive of all clients :)

File: roadmap.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+Note: Although critical to the OE roadmap, this project is tracked [external](https://github.com/ethereum) to this repository.
+
+### eth2
+
+The eth2 merge API and sharding will allow for native integration into execution engines (eg. Geth) as well as massive scalability ([~100k TPS](https://vitalik.ca/general/2021/01/05/rollup.html)) by greatly increasing Ethereum's data availability bandwidth.
```
Comment (lightclient) [inline roadmap.md:38]:
```suggestion
The eth2 merge API and sharding will allow for native integration into execution engines as well as massive scalability ([~100k TPS](https://vitalik.ca/general/2021/01/05/rollup.html)) by greatly increasing Ethereum's data availability bandwidth.
```

File: roadmap.md (hunk 4 above)
Comment (karlfloersch) [reply roadmap.md:32]: Oh you are totally right. I need to break my Geth bias lol thank you

File: roadmap.md (hunk 5 above)
Comment (karlfloersch) [reply roadmap.md:38]: That geth bias tho

File: roadmap.md (hunk 1 above)
Comment (karlfloersch) [reply roadmap.md]: Agreed 100%. Updated the language! This is going to have to be hella stable, you are 100% correct fam.

**Comment by tynes** [comment]
Are the SVG files rendering for you? They just show up as white rectangles for me

**Comment by karlfloersch** [comment]
Hmm... @tynes I think it might be a glitch in the Git diff because it's showing up fine in the readme.

**Comment by karlfloersch** [comment]
Going to merge this for now and we can update it over time

**Pull request merged**